library/core/src/iter/traits/iterator.rs

```diff
@@ -2998,6 +2998,11 @@ pub trait Iterator {
     /// This is useful when you have an iterator over `&T`, but you need an
     /// iterator over `T`.
     ///
+    /// Note that unlike copying, cloning is regarded as a side effect, because
+    /// `Clone` implementations may contain observable behavior, so for example
+    /// on `iter.cloned().last()`, all elements will be cloned even if only the
```

Comment:
This sentence is a bit of a run-on:

```suggestion
    /// `Clone` implementations may contain observable behavior. So, for example
    /// with `iter.cloned().last()`, all elements will be cloned even if only the
```

Reply:
I split it off into its own sentence.